File: ubuntu/WallpaperInfoApp/WallpaperServiceConnection.py
#!/usr/bin/env python3

import WallpaperServiceHandler
import WallpaperServiceInfo as w
import WallpaperServiceConnection

class WallpaperServiceConnection:

	def __init__(self, wallpaperInfoUI):
		self._wallpaperInfoUI = wallpaperInfoUI
		self._wallpaperServiceHandler = None

	def sendMessageToService(self, command, intOption = 0, objectOption = None):
		if self._wallpaperServiceHandler != None:
			self._wallpaperServiceHandler.handleMessage(command, intOption, objectOption)

	def unbindService(self):
		self._wallpaperServiceHandler.unbind()
		self._wallpaperServiceHandler = None

def startService():
	WallpaperServiceHandler.startService()

def stopService():
	WallpaperServiceHandler.stopService()

def broadcastToService(action, extra = 0):
	w.WallpaperServiceInfo().getWallpaperService().broadcastReceiver(action, extra)

def bindService(wallpaperInfoUI):
	wallpaperServiceConnection = WallpaperServiceConnection(wallpaperInfoUI)
	wallpaperServiceConnection._wallpaperServiceHandler = WallpaperServiceHandler.incomingHandler(wallpaperServiceConnection)
	return wallpaperServiceConnection

File: apps/desktop-macosx/WallpaperInfoApp/WallpaperInfoApp/WallpaperServiceConnection.swift
//
//  WindowServiceConnection.swift
//  WallpaperInfoApp
//
//  Created by Brad Park on 4/15/20.
//  Copyright © 2020 Brad Park. All rights reserved.
//

import Foundation

class WallpaperServiceConnection {
	
	init(_ wallpaperInfoUI: WallpaperInfoUI) {
		_delegate = wallpaperInfoUI
	}
	
    public static func startService() {
        WallpaperServiceHandler.startService()
    }
    
    public static func stopService() {
        WallpaperServiceHandler.stopService()
    }
    
	public static func broadcastToService(_ action:Int, extras:Any? = nil) {
        WallpaperServiceInfo.getInstance().getWallpaperService().broadcastReceiver(action, extras)
    }
	
    func sendMessageToService(command:Int, intOption:Int = 0, objectOption:Any? = nil) {
		if _wallpaperServiceHandler != nil {
			_wallpaperServiceHandler?.handleMessage(command, intOption, objectOption)
		}
    }
	
	class func bindService(WallpaperInfoWindowController: WallpaperInfoUI) -> WallpaperServiceConnection {
		let wallpaperServiceConnection = WallpaperServiceConnection(WallpaperInfoWindowController)
		wallpaperServiceConnection._wallpaperServiceHandler = WallpaperServiceHandler.incomingHandler(incomingMessenger: wallpaperServiceConnection)
		return wallpaperServiceConnection
	}
	
	func unbindService() {
		_wallpaperServiceHandler?.unbind()
		_wallpaperServiceHandler = nil
	}
	
	var _delegate: WallpaperInfoUI
	private var _wallpaperServiceHandler: WallpaperServiceHandler? = nil
}
